### ISSUE TITLE
Info about dynamic components

### ISSUE THREAD

**Issue by d3caf** [surface-ui/surface#556]
Hello! Thanks so much for all the hard work on this!

I was wondering if there was someplace to see more detail on how to use either `Dynamic.Component` or `Dynamic.LiveComponent`. I've struggling to figure out the best way to implement it. I'm attempting to pass in sort of a "render prop" (coming from a React background) to show a custom element in my table header. I stole the markup from the Surface sample table where it uses a renderless component for each column.

Currently, I have it rendering, but am a bit confused about how I pass assigns to the `header/1` function. Perhaps I'm using this wrong, I'm not sure.

Thanks again!

```elixir
# --- Implementation
def render(assigns) do
    ~F"""
    <SimpleTable data={stat <- @doc} light>
      <SimpleTable.Column align={:left}><span title={stat.description}>{stat.label}</span></SimpleTable.Column>
      <SimpleTable.Column label={header(%{})}><Badge value={stat.value} /></SimpleTable.Column>
    </SimpleTable>
    """
end
```
```elixir
def header(assigns) do
      <Badge large value="75" />
end
```
```elixir
# --- Component
        <thead>
          <tr>
            <th :for={col <- @cols} style={css("text-align": col.align)}>
              {#if is_function(col.label)}
                <Dynamic.Component module={col.label} />
              {#else}
                <Icon :if={col.icon_name} name={col.icon_name} />
                {col.label}
              {/if}
            </th>
          </tr>
        </thead>

```

**Comment by LeakyBucket** [comment]
A little late to the party and I'm not sure it would solve your issue entirely but after messing around with trying to figure out how to send props _through_ a Dynamic Component I landed on the following which appears to work (at least in 0.7.4):

```
<Field class="grow" name={@field.name}>
  <Label>{label(@field)}</Label>
  <div>
    <Context put={Field, field: @field.name}>
      {#if is_live_component?(@field)}
        <LiveComponent module={field_component(@field)} {...props(@field)} id={@field.name} />
      {#else}
        <Component module={field_component(@field)} {...props(@field)} />
      {/if}
    </Context>
  </div>
</Field>
```

It turns out there are issues with the Context not passing properly with dynamic components however the props/assigns all come through as expected as long as the `props/1` function returns a map with the intended prop names as keys and the values as expected by the "concrete" component being rendered.

**Comment by msaraiva** [comment]
Fixed in [e8058cd5fa4d5c684743325f7e8a850437bd571f](https://github.com/surface-ui/surface_site/commit/e8058cd5fa4d5c684743325f7e8a850437bd571f).
